internal/drivers/google/driver.go

```diff
@@ -539,8 +540,15 @@ func (p *config) findInstanceZone(ctx context.Context, instanceID string) (
 
 func (p *config) waitZoneOperation(ctx context.Context, name, zone string) error {
 	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		ctxGcp, cancel := context.WithTimeout(ctx, operationGetTimeout*time.Second)
 		client := p.service
-		op, err := client.ZoneOperations.Get(p.projectID, zone, name).Context(ctx).Do()
+		op, err := client.ZoneOperations.Get(p.projectID, zone, name).Context(ctxGcp).Do()
+		cancel()
```

Comment:
good catch on using `cancel` since it's in an infinite for loop

Reply:
Yeah linter helped here :)